packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -141,13 +131,18 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    final FlutterCommandResult xcarchiveResult = await super.runCommand();
-    final BuildInfo buildInfo = await getBuildInfo();
+    final BuildInfo buildInfo = await cachedBuildInfo;
```

Comment:
This should have been the `late final` `cachedBuildInfo` so `getBuildInfo()` isn't called twice.